packages/cw721/src/execute.rs

```diff
@@ -502,22 +221,293 @@ where
+    let extension_attributes = collection_info.extension.to_attributes_state()?;
+    config
+        .collection_info
+        .save(deps.storage, &collection_info.into())?;
+    for attr in extension_attributes {
+        config
+            .collection_extension
+            .save(deps.storage, attr.key.clone(), &attr)?;
+    }
+
+    let response = Response::new().add_attribute("action", "update_collection_info");
+    if let Some(info) = info {
+        Ok(response.add_attribute("sender", info.sender.to_string()))
+    } else {
+        Ok(response)
+    }
+}
+
+#[allow(clippy::too_many_arguments)]
+pub fn mint<TNftExtension, TNftExtensionMsg, TCustomResponseMsg>(
+    deps: DepsMut,
+    env: &Env,
+    info: &MessageInfo,
+    token_id: String,
+    owner: String,
+    token_uri: Option<String>,
+    extension: TNftExtensionMsg,
+) -> Result<Response<TCustomResponseMsg>, Cw721ContractError>
+where
+    TNftExtension: Cw721State,
+    TNftExtensionMsg: Cw721CustomMsg + StateFactory<TNftExtension>,
+    TCustomResponseMsg: CustomMsg,
+{
+    // create the token
+    let token_msg = NftInfoMsg {
+        owner: owner.clone(),
+        approvals: vec![],
+        token_uri: token_uri.clone(),
+        extension,
+    };
+    let token = token_msg.create(deps.as_ref().into(), env.into(), info.into(), None)?;
+    let config = Cw721Config::<TNftExtension>::default();
+    config
+        .nft_info
+        .update(deps.storage, &token_id, |old| match old {
+            Some(_) => Err(Cw721ContractError::Claimed {}),
+            None => Ok(token),
+        })?;
+
+    config.increment_tokens(deps.storage)?;
+
+    let mut res = Response::new()
+        .add_attribute("action", "mint")
+        .add_attribute("minter", info.sender.to_string())
+        .add_attribute("owner", owner)
+        .add_attribute("token_id", token_id);
+    if let Some(token_uri) = token_uri {
+        res = res.add_attribute("token_uri", value_or_empty(&token_uri));
+    }
+    Ok(res)
+}
+
+pub fn update_minter_ownership<TCustomResponseMsg>(
+    api: &dyn Api,
+    storage: &mut dyn Storage,
+    env: &Env,
+    info: &MessageInfo,
+    action: Action,
+) -> Result<Response<TCustomResponseMsg>, Cw721ContractError> {
+    let ownership = MINTER.update_ownership(api, storage, &env.block, &info.sender, action)?;
+    Ok(Response::new()
+        .add_attribute("update_minter_ownership", info.sender.to_string())
+        .add_attributes(ownership.into_attributes()))
+}
+
+pub fn update_creator_ownership<TCustomResponseMsg>(
+    api: &dyn Api,
+    storage: &mut dyn Storage,
+    env: &Env,
+    info: &MessageInfo,
+    action: Action,
+) -> Result<Response<TCustomResponseMsg>, Cw721ContractError> {
+    let ownership = CREATOR.update_ownership(api, storage, &env.block, &info.sender, action)?;
+    Ok(Response::new()
+        .add_attribute("update_creator_ownership", info.sender.to_string())
+        .add_attributes(ownership.into_attributes()))
+}
```

Comment:
`update_minter_ownership` and `update_creator_ownership` uses cw_ownable for transferring ownership!